cli/js/io.ts

```diff
@@ -110,3 +112,46 @@ export function* iterSync(
     yield b.subarray(0, result);
   }
 }
+
+export function toWriter(
+  streamWriter: WritableStreamDefaultWriter<Uint8Array>
+): Writer {
+  return {
+    async write(p: Uint8Array): Promise<number> {
+      await streamWriter.ready;
+      await streamWriter.write(p);
+      return p.length;
+    },
+  };
+}
+
+export function toReader(
+  streamReader: ReadableStreamDefaultReader<Uint8Array | string>
+): Reader {
+  const encoder = new TextEncoder();
+  const buffer = new Buffer();
+
+  return {
+    async read(p: Uint8Array): Promise<number | null> {
+      if (buffer.empty()) {
+        const res = await streamReader.read();
+        if (res.done) {
+          return null; // EOF
+        }
+
+        let value = res.value;
+        if (!value) {
+          return 0;
+        }
```

Comment:
This check shouldn't be needed according to the types.

Reply:
empty string for example. And it can be used from JavaScript code, so someone can enqueue `null` or `undefined`. If we want to throw an error if someone enqueues `undefined` or `null` I can improve the check to: `value === ''`.
But I don't think we should throw if an empty chunk is pushed.

Reply:
> empty string for example. And it can be used from JavaScript code, so someone can enqueue `null` or `undefined`. If we want to throw an error if someone enqueues `undefined` or `null` I can improve the check to: `value === ''`.
> But I don't think we should throw if an empty chunk is pushed.

An empty `Uint8Array` would be handled naturally by the `writeAll()`. The others will throw type errors :) As for JS, we don't do runtime type checking. Wrong types are just undefined behaviour but it'll usually hit a `ReferenceError` somwhere.

Reply:
The check was done for empty string, an empty `Uint8Array` won't match that condition. In any case I'll remove it.